.github/workflows/test.yml

```diff
@@ -12,30 +12,30 @@ jobs:
 
     # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
-    # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-    - uses: actions/checkout@v4
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+    - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
```

Comment:
Is `@4` not enough? Do you think it risks breaking in the future?

Reply:
I guess for generated files, it might be reasonable not to pin them, but for us, it's more secure not to.
Pinning to a commit means that a malicious actor can't change the tag, as a commit is immutable, but not a tag. GitHub also doesn't have the greatest track record with both bugs and SemVer, and has happily broken the latest versions of actions for weeks at a time.
Dependabot should bump them, so missing patches shouldn't be a major issue.

I explained this a few weeks ago in a similar PR, so if my wording doesn't make sense (I personally doubt I said this in a way that makes sense), try this explanation instead: <https://github.com/material-extensions/vscode-material-icon-theme/pull/2365#discussion_r1639112514>.

Reply:
What do you think about adding pinning these versions in this file and in `new-package` generated files, and adding a comment to explain the reason and how to upgrade (how/where to find the new commit for instance). I think that way both this project and users of the package benefit from the improvements here.

Reply:
Yeah. That was a lot more noise than I'd expected.

Reply:
> Yeah. That was a lot more noise than I'd expected.

Same here :sweat_smile: 
The GitHub rate limit has exploded :joy: 

> With elm packages though, it should just notify that there're new majors.

Hmm, I don't even know if it should. For a package author, as long as they can run tools and tests (locally and in CI) and publish (through CI), they don't need to have the latest version of anything :man_shrugging:

Reply:
Yeah, but it would block the application from upgrading other review rules.
Say `elm-review-simplify` needs `elm-review` 2 or 3, but only declares support for 2, and `elm-review-unused` needs `elm-review` 3, then one couldn't get the improvements to `elm-review-unused`.

Reply:
Another thing, I think I forgot to set it to weekly, I usually have it set to daily, but weekly is a lot more manageable in my experience.